geo/src/algorithm/bool_ops/tests.rs

```diff
@@ -1,8 +1,44 @@
 use super::{unary_union, BooleanOps};
 use crate::{wkt, Convert, MultiPolygon, Polygon, Relate};
 use std::time::Instant;
+use i_overlay::core::fill_rule::FillRule;
 use wkt::ToWkt;
 
+#[test]
+fn test_union_fill_rules() {
+    let self_intersecting_star: Polygon = wkt!(POLYGON((50.0 0.0, 21.0 90.0, 98.0 35.0, 2.0 35.0, 79.0 90.0 )));
+    let square: Polygon = wkt!(POLYGON((50.0 50.0, 94.0 50.0, 94.0 8.0, 50.0 8.0)));
+
+    let multi1 = self_intersecting_star.union(&square);
+    assert_eq!(multi1.0.len(), 4);
+
+    let multi2 = self_intersecting_star.union_with_fill_rule(&square, FillRule::EvenOdd);
+    assert_eq!(multi2.0.len(), 4);
+    assert_eq!(multi1, multi2);
+
+    let additive_union = self_intersecting_star.union_with_fill_rule(&square, FillRule::NonZero);
+    assert_eq!(additive_union.0.len(), 1);
+}
+
+#[test]
+fn test_difference_fill_rules() {
+    let self_intersecting_star: Polygon = wkt!(POLYGON((50.0 0.0, 21.0 90.0, 98.0 35.0, 2.0 35.0, 79.0 90.0)));
+    let square: Polygon = wkt!(POLYGON((50.0 50.0, 94.0 50.0, 94.0 8.0, 50.0 8.0)));
+
+    let multi1 = self_intersecting_star.difference(&square);
+    assert_eq!(multi1.0.len(), 6);
+
+    let multi2 = self_intersecting_star.difference_with_fill_rule(&square, FillRule::EvenOdd);
+    assert_eq!(multi2.0.len(), 6);
+    assert_eq!(multi1, multi2);
+
+    println!("multi1: {}", multi1.wkt_string());
```

Comment:
Please remove these `println!`

Reply:
Ah good catch again. I was using that to generate the images.